ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -5892,54 +5940,73 @@ public CheckConstraint getCheckConstraints(String dbName, String tblName)
   public void addPrimaryKey(List<SQLPrimaryKey> primaryKeyCols)
     throws HiveException, NoSuchObjectException {
     try {
+      Table tbl = getTable(primaryKeyCols.get(0).getTable_db(), primaryKeyCols.get(0).getTable_name());
+
```

Comment:
Why is this needed?

Reply:
Rebased with master and only kept relevant changes.